modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/storage/sql/TestS3DirectBatchHandler.java

```diff
@@ -162,6 +164,51 @@ public void test20MB() throws SdkBaseException, InterruptedException {
         test("s3", 20 * 1024 * 1024);
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.core.storage.binarymanager.s3.tests:OSGI-INF/test-s3directupload-contrib.xml")
+    public void testTokenRenewal() throws SdkBaseException, InterruptedException {
+        // Test that refreshing tokens actually returns back valid and different tokens.
+        S3DirectBatchHandler handler = (S3DirectBatchHandler) batchManager.getHandler("s3");
+        Batch newBatch = handler.newBatch(null);
+        Batch batch = handler.getBatch(newBatch.getKey());
+
+        // Save current details
+        Map<String, Object> properties = batch.getProperties();
+        String secretKeyId = (String) properties.get(S3DirectBatchHandler.INFO_AWS_SECRET_KEY_ID);
+        String secretAccessKey = (String) properties.get(S3DirectBatchHandler.INFO_AWS_SECRET_ACCESS_KEY);
+        String sessionToken = (String) properties.get(S3DirectBatchHandler.INFO_AWS_SESSION_TOKEN);
+        Long expiration = (Long) properties.get(S3DirectBatchHandler.INFO_EXPIRATION);
+
+        // Wait for some time else the new expiration will not be updated
+        Thread.sleep(1000); // 1s
```

Comment:
Better to use awaitility instead, see https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-platform-tag-service/nuxeo-platform-tag-core/src/test/java/org/nuxeo/ecm/platform/tag/AbstractTestTagService.java#L120 for instance.

Reply:
I finally removed that logic, it would introduce random failures. The new check is less error prone.